llvm/docs/LangRef.rst

```diff
@@ -19644,13 +19644,18 @@ in order to get the number of elements to process on each loop iteration. The
 result should be used to decrease the count for the next iteration until the
 count reaches zero.
 
-If the count is larger than the number of lanes in the type described by the
-last 2 arguments, this intrinsic may return a value less than the number of
-lanes implied by the type. The result will be at least as large as the result
-will be on any later loop iteration.
-
-This intrinsic will only return 0 if the input count is also 0. A non-zero input
-count will produce a non-zero result.
+Let ``%max_lanes`` be the number of lanes in the type described by ``%vf`` and
+``%scalable``, here are the constraints on the returned value:
+- If ``%cnt`` equals to 0, returns 0.
+- The returned value is always less or equal to ``%max_lanes``.
+- The returned value is always larger or equal to ``ceil(%cnt / ceil(%cnt / %max_lanes))``.
+  - This implies that if ``%cnt`` is non-zero, the result should be non-zero
+    as well.
+  - This also implies that if ``%cnt`` is less than ``%max_lanes``, it has to
```

Comment:
The way we're indenting these implications says to me (on first reading) that the implications fall out of only the last equation, when in fact it's a consequence of a combination of all of them.

Could we perhaps list the three conditions then introduce the implications with a new set of bullet points?

```
.. on the returned value:

- If %cnt is 0, returns 0
- <= %max_lanes
- >= ceil(...)

Note that this has the following implications:

- If %cnt is non-zero, the result should be non-zero as well
- If %cnt is less than %max_lanes, returns %cnt.
```

Reply:
Agree. It's updated.